### PR TITLE
Hts 674

### PR DESCRIPTION
**Pull request by Jackie1988**

### CODE REVIEW

**Comment by AndyHWChung** [comment]
do we want to follow the directory structure followed by other apis?

https://github.com/hmrc/estate-registration-api/tree/master/resources/public/api/conf/1.0
https://github.com/hmrc/lisa-api/tree/master/resources/public/api/conf/1.0
https://github.com/hmrc/individuals-api/tree/master/resources/public/api/conf/1.0

We might not need to, but it makes sense to have all our documents together when we eventually have all of them

**Comment by Jackie1988** [comment]
ok thanks Andy, I have now created the same directory structure as those repos above

File: resources/public/api/conf/1.0/application.raml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  headers: http://api-documentation-raml-frontend.service/api-documentation/assets/common/modules/headers.raml
+  annotations: http://api-documentation-raml-frontend.service/api-documentation/assets/common/modules/annotations.raml
+  types: http://api-documentation-raml-frontend.service/api-documentation/assets/common/modules/types.raml
+
+/individuals/help-to-save/account:
+  post:
+    description: Send a request to create an account from KCOM on behalf of the user
+    body:
+      application/json:
+        schema: |
```
Comment (AndyHWChung) [inline resources/public/api/conf/1.0/application.raml]: this looks like an example rather than a schema. Do we want a schema and then an example?

**Comment by Jackie1988** [comment]
ok thanks for the comments, I have added http header validation and timeouts, but I don't know how to put in the data transformations

**Comment by Jackie1988** [comment]
I've added the data transformations in the form of written text included in the raml doc via a markdown text file

File: resources/public/api/conf/1.0/docs/errors.md [hunk 2]
```diff
@@ -0,0 +1 @@
+2xx range indicates success, 4xx and 5xx ranges indicates failure
```
Comment (AndyHWChung) [inline resources/public/api/conf/1.0/docs/errors.md]: Is it worth saying here 201 is the expected response and 409 is also not really an error but indicates an account has already been created. All other responses are then errors

File: resources/public/api/conf/1.0/application.raml [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+               "communicationPreference": "00" 
+              }, 
+             "registrationChannel": "callCentre" 
+           }
+          }
+    responses:
+      201:
+        body:
+          application/json:
+            type: !include schemas/CreateAccount.json
```
[the remark this answers is not in the document]
Comment (AndyHWChung) [reply resources/public/api/conf/1.0/application.raml]: also, do we want a textual description of the schema before we give the actual schema as in the spec?

[the remark this answers is not in the document]
Comment (Jackie1988) [reply resources/public/api/conf/1.0/application.raml]: Yes sorry I've now placed the schema in the correct place. What the schema is should be obvious, as theres a description under post.

File: resources/public/api/conf/1.0/docs/errors.md (hunk 2 above)
Comment (Jackie1988) [reply resources/public/api/conf/1.0/docs/errors.md]: Yes good idea, I will add that.

**Comment by Jackie1988** [comment]
ok thanks for the comments. I have already included the timeouts near the top. And I have added access to the API, diacritics and legal considerations and security matters as documents.

File: resources/public/api/conf/1.0/docs/access.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+The caller is expected to invoke a private HMRC API Platform endpoint using OAuth 2.0 & OpenID Connect, as documented
+here: https://developer.service.hmrc.gov.uk/api-documentation/docs/authorisation/application-restricted-endpoints
+
+The vendor MUST create a new Developer hub account and MUST request a new ApplicationID from HMRC for using this API as
+the API permissioning is made against this ID. Perform the following steps:
+
+#### Initial Testing
+Step 1: setup a new Sandbox Test account at https://test-developer.service.hmrc.gov.uk/api-documentation old accounts must not be reused.
```
Comment (AndyHWChung) [inline resources/public/api/conf/1.0/docs/access.md]: It'll look nicer if the steps are in a list: see here for creating lists in markdown https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#lists

File: resources/public/api/conf/1.0/docs/access.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+Step 2: register the application at https://test-developer.service.hmrc.gov.uk/developer/login as new in order to acquire an ApplicationID.
+Step 3. Send the ApplicationID + “test” environment name to the Help To Save Dev Team via an email to
+hmrc-mdtp-team-helptosave-g@digital.hmrc.gov.uk whereby they raise an internal ticket onto the API Platform backlog to
+setup access, and will additionally specify ‘grant suppress’ as part of this ticket. Once this ticket is done, the Help
+To Save Dev Team will notify the vendor that the API is ready to “sandbox test”.
+
+#### Live running
+Step 1. setup a Live account at https://developer.service.hmrc.gov.uk/api-documentation old accounts must not be reused.
```
Comment (AndyHWChung) [inline resources/public/api/conf/1.0/docs/access.md]: a list would be nice here as well

File: resources/public/api/conf/1.0/docs/dataTransformations.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+#### Data Transformations by HMRC:
+
+For the following fields; forename, surname, address1, address2, address3, address4, address5 and phoneNumber,
```
Comment (AndyHWChung) [inline resources/public/api/conf/1.0/docs/dataTransformations.md]: a list might make things clearer here

File: resources/public/api/conf/1.0/docs/dataTransformations.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+control characters TAB \t, NEWLINE \n, CR \r are replaced with single space, consecutive spaces greater than 1
+are replaced with a single space and leading & trailing spaces are removed.
+
+#### Data Transformations by NSI:
+
+For the following fields; forename, surname, address1, address2, address3, address4, address5 and phoneNumber,
```
Comment (AndyHWChung) [inline resources/public/api/conf/1.0/docs/dataTransformations.md]: a list might make things more clear here as well

File: resources/public/api/conf/1.0/application.raml [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+    headers:
+      content-type: application/json
+      authorization: bearer [YOUR-SERVER-TOKEN]
+      gov-client-user-ID: EL079651A
+      gov-client-timezone: UTC
+      gov-vendor-version: 1.3
+      gov-vendor-instance-ID: fa7c484c-d8cf-4f8a-a39c-54a2a4c04bc6
+    body:
+      application/json:
+        schema: !include schemas/CreateAccount.json
```
Comment (AndyHWChung) [inline resources/public/api/conf/1.0/application.raml]: I can't seem to find the schema in the generated HTML - is it just me?

File: resources/public/api/conf/1.0/docs/access.md (hunk 4 above)
Comment (Jackie1988) [reply resources/public/api/conf/1.0/docs/access.md]: ok thanks, thats now done

File: resources/public/api/conf/1.0/docs/access.md (hunk 5 above)
Comment (Jackie1988) [reply resources/public/api/conf/1.0/docs/access.md]: done

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**